src/modules/cmdNotFound/CmdNotFoundModuleInterface/dllmain.cpp

```diff
@@ -0,0 +1,143 @@
+// dllmain.cpp : Defines the entry point for the DLL application.
+#include "pch.h"
+
+#include <filesystem>
+#include <string>
+
+#include <common/logger/logger.h>
+#include <common/logger/logger_settings.h>
+#include <common/SettingsAPI/settings_objects.h>
+#include <common/utils/gpo.h>
+#include <common/utils/logger_helper.h>
+#include <common/utils/process_path.h>
+#include <common/utils/resources.h>
+#include <interface/powertoy_module_interface.h>
+
+#include "resource.h"
+#include "trace.h"
+
+BOOL APIENTRY DllMain(HMODULE /*hModule*/, DWORD ul_reason_for_call, LPVOID /*lpReserved*/)
+{
+    switch (ul_reason_for_call)
+    {
+    case DLL_PROCESS_ATTACH:
+        Trace::RegisterProvider();
+        break;
+    case DLL_THREAD_ATTACH:
+    case DLL_THREAD_DETACH:
+        break;
+    case DLL_PROCESS_DETACH:
+        Trace::UnregisterProvider();
+        break;
+    }
+    return TRUE;
+}
+
+const static wchar_t* MODULE_NAME = L"Command Not Found";
+const static wchar_t* MODULE_DESC = L"A module that detects an error thrown by a command in PowerShell and suggests a relevant WinGet package to install, if available.";
+
+inline const std::wstring ModuleKey = L"CmdNotFound";
+
+class CmdNotFound : public PowertoyModuleIface
+{
+    std::wstring app_name;
+    std::wstring app_key;
+
+private:
+    bool m_enabled = false;
+
+    void install_module()
+    {
+        auto module_path = get_module_folderpath();
+
+        std::string command = "pwsh.exe";
+        command += " ";
+        command += "-NoProfile -NonInteractive -NoLogo -WindowStyle Hidden -ExecutionPolicy Unrestricted -File \"" + winrt::to_string(module_path) + "\\WinUI3Apps\\Assets\\Settings\\Scripts\\EnableModule.ps1" + "\"" + " -scriptPath \"" + winrt::to_string(module_path) + "\"";
+
+        system(command.c_str());
```

Comment:
Is this going to crash if for example it fails while running? (pwsh not installed, for example)
Thinking we might want to add some logs around this?

Reply:
Won't crash, it'll return non-zero value if the system() fails for whatever reason (including pwsh.exe not found). Added check and logs.